python/lsst/pipe/analysis/fakesAnalysis.py

```diff
@@ -988,91 +1142,28 @@ def fakesCompletenessPlot(inputFakes, inputFakesMatched, processedFakesMatched,
 
     band = plotInfoDict["filter"][-1].lower()
 
-    # Make a color map that looks nice
-    r, g, b = colors.colorConverter.to_rgb("C0")
-    r1, g1, b1 = colors.colorConverter.to_rgb("midnightblue")
-    colorDict = {"blue": ((0.0, b, b), (1.0, b1, b1)), "red": ((0.0, r, r), (1.0, r1, r1)),
-                 "green": ((0.0, g, g), (1.0, g1, g1))}
-    colorDict["alpha"] = ((0.0, 0.2, 0.2), (0.05, 0.3, 0.3), (0.5, 0.8, 0.8), (1.0, 1.0, 1.0))
-    newBlues = colors.LinearSegmentedColormap("newBlues", colorDict)
-
-    # Find the fake stars"
+    # Find the fake stars
     stars = np.where((inputFakes["sourceType"] == "star"))[0]
-    starsMatched = np.where((inputFakesMatched["sourceType"] == "star"))[0]
-
-    # Find the min/max ra/dec and use them to find an approximate scale for the plot limits.
-    minRa = np.min(inputFakesMatched[raFakesCol].values[starsMatched])
-    maxRa = np.max(inputFakesMatched[raFakesCol].values[starsMatched])
-    minDec = np.min(inputFakesMatched[decFakesCol].values[starsMatched])
-    maxDec = np.max(inputFakesMatched[decFakesCol].values[starsMatched])
-
-    scaleRa = np.fabs((maxRa - minRa) / 20.0)
-    scaleDec = np.fabs((maxDec - minDec) / 20.0)
 
-    rasFakes = inputFakes[raFakesCol].values[stars]
-    decsFakes = inputFakes[decFakesCol].values[stars]
+    def histFunction(zs, plotLimits=False):
+        zsOut = 1.0 - (np.sum(zs)/len(zs))
 
-    rasFakesMatched = inputFakesMatched[raFakesCol].values[starsMatched]
-    decsFakesMatched = inputFakesMatched[decFakesCol].values[starsMatched]
+        if len(zs) == 0:
+            zsOut = np.nan
 
-    fig, ax = plt.subplots()
-
-    # Run through the ccds, plot them and then histogram the data on them
-    for ccd in list(set(processedFakesMatched["ccdId"].values)):
-        # Find the ccd corners and sizes
-        corners = areaDict["corners_" + str(ccd)]
-        xy = (corners[0].getRa().asDegrees(), corners[0].getDec().asDegrees())
-        width = corners[1].getRa().asDegrees() - corners[0].getRa().asDegrees()
-        height = corners[1].getDec().asDegrees() - corners[0].getDec().asDegrees()
-
-        # Some of the ccds are rotated and some have xy0 as being on the right with negative width/height
-        # this upsets the binning so find the min and max to calculate positive bin widths from.
-        minX = np.min([xy[0], corners[1].getRa().asDegrees()])
-        maxX = np.max([xy[0], corners[1].getRa().asDegrees()])
-        minY = np.min([xy[1], corners[1].getDec().asDegrees()])
-        maxY = np.max([xy[1], corners[1].getDec().asDegrees()])
-        if np.fabs(width) > np.fabs(height):
-            binWidth = (maxX - minX) / 10
+        if plotLimits:
+            return 0, 1
```

Comment:
So this sometimes returns a tuple and sometimes returns a scalar?? Your use of this on line 865 won't work if it returns a scalar or a nan: 
`        vmin, vmax = statistic(zs, plotLimits=True)`

Reply:
I have no idea what was going on here. I have got rid of the plotLimits bit, it isn't used anywhere as far as I can see. I'll check it all works still.